daemon/src/wire.rs

```diff
@@ -109,6 +130,30 @@ pub mod maker_to_taker {
         Confirm,
         Reject,
     }
+
+    /// A generic string message that can be sent from maker to taker
+    ///
+    /// This can be used to e.g. warn the taker user that a version will be deprecated and an
+    /// upgrade is coming. This message is not strongly typed on purpose to keep it configurable
+    /// for the maker. This message should only be used very mindfully.
+    #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+    pub struct Alarm {
+        pub message: String,
+    }
+
+    impl fmt::Display for Alarm {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            self.message.fmt(f)
+        }
+    }
+
+    /// A fixed set of goodbye messages from maker to taker
+    ///
+    /// This helps the taker to understand the reason why a connection was closed.
+    #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+    pub enum Goodbye {
+        VersionMismatch { maker_version: Version },
```

Comment:
I think this expresses the semantics a bit better.

```suggestion
        UnsupportedVersion { minimum_version: Version },
```

Reply:
I think it could happen that a maker did not uppgrade, and the taker is one version ahead. That's why I opted for the `maker_version` instead of minimum. Minimum also makes it harder to decide what that means - I think if we want that we would have to use semantic versioning already, so we can decide what "minimum supported version" means.